File: src/main/java/alfheim/common/potion/PotionBleeding.java
package alfheim.common.potion;

import alfheim.AlfheimCore;
import alfheim.common.core.util.AlfheimConfig;
import alfheim.common.core.util.DamageSourceSpell;
import net.minecraft.entity.EntityLivingBase;

public class PotionBleeding extends PotionAlfheim {

	public PotionBleeding() {
		super(AlfheimConfig.potionIDBleeding, "bleeding", true, 0xFF0000);
	}

	public boolean isReady(int time, int ampl) {
		return time % (20/Math.max(1, ampl)) == 0; 
	}

	public void performEffect(EntityLivingBase living, int ampl) {
		if (AlfheimCore.enableMMO) living.attackEntityFrom(DamageSourceSpell.bleeding, ampl+1);
	}
}

File: src/main/java/alfheim/common/potion/PotionBleeding.kt
package alfheim.common.potion

import alfheim.AlfheimCore
import alfheim.common.core.handler.AlfheimConfigHandler
import alfheim.common.core.util.DamageSourceSpell
import net.minecraft.entity.EntityLivingBase
import kotlin.math.max

class PotionBleeding: PotionAlfheim(AlfheimConfigHandler.potionIDBleeding, "bleeding", true, 0xFF0000) {
	
	override fun isReady(time: Int, ampl: Int) = time % (20 / max(1, ampl)) == 0
	
	override fun performEffect(living: EntityLivingBase, ampl: Int) {
		if (AlfheimCore.enableMMO) living.attackEntityFrom(DamageSourceSpell.bleeding, (ampl + 1).toFloat())
	}
}
